### PR TITLE
Add multithread support

### PR DESCRIPTION
**Pull request by li-wu**

### CODE REVIEW

**Comment by arctan5x** [comment]
Hi, we also have a logic in setup_http function to update threading option to process.

File: splunk_eventgen/eventgen_api_server/eventgen_core_object.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def _create_args(self):
         args.interval = None
         args.keepoutput = False
         args.modinput = False
-        args.multiprocess = True
+        args.multiprocess = False if kargs.get("mutithread") else True
```
Comment (arctan5x) [inline splunk_eventgen/eventgen_api_server/eventgen_core_object.py]: typo

Comment (li-wu) [reply splunk_eventgen/eventgen_api_server/eventgen_core_object.py]: Fixed.

**Comment by li-wu** [comment]
> setup_http

you mean the logic here?
https://github.com/splunk/eventgen/blob/develop/splunk_eventgen/eventgen_api_server/eventgen_server_api.py#L536

**Comment by arctan5x** [comment]
@li-wu Yes. That is correct.

**Comment by li-wu** [comment]
@arctan5x I walked through the function `setup_http` and it seems the `threading` is set to `process` and no way to change it to `thread`.  Any thing missing?

**Comment by li-wu** [comment]
@arctan5x could you help confirm my previous comment? Thanks.

**Comment by arctan5x** [comment]
What do you mean by that there is no way to change it to "thread"?

**Comment by arctan5x** [comment]
Seems like that there is no dynamic way to change threading mode while the server is running. Since users don't control how server starts (only able to control threading through args/evn vars), how are users using this change?

**Comment by li-wu** [comment]
> What do you mean by that there is no way to change it to "thread"?

I mean there is no logic to change the mode to thread mode in the function `setup_http `.

**Comment by li-wu** [comment]
> Seems like that there is no dynamic way to change threading mode while the server is running. Since users don't control how server starts (only able to control threading through args/evn vars), how are users using this change?

If users like perf team are using the docker, that might be a problem since they are using /start REST endpoint to start the service. Any good suggestion?

**Comment by Yangxulight** [comment]
It's ok for our perf team. Actually, we only need to make sure the eventgen server mode using multithread by default. I think we can change the `entrypoint.sh` with `--multithread` parameter on server mode when we want to upload a new image.

**Comment by arctan5x** [comment]
If perf team is going to change entrypoint.sh, I would say this PR would do its job. One thing I would note is that /setup changes threading to multiprocess. So please note that when you use.

**Pull request merged**